core/Makefile

```diff
@@ -12,15 +12,19 @@ DEPFLAGS = -MT $@ -MMD -MP -MF $(DEPDIR)/$(@:.o=.d)
 $(shell mkdir -p $(DEPDIR) >/dev/null)
 $(shell mkdir -p $(DEPDIR)/utils >/dev/null)
 $(shell mkdir -p $(DEPDIR)/modules >/dev/null)
-$(shell mkdir -p $(DEPDIR)/hooks >/dev/null)
 $(shell mkdir -p $(DEPDIR)/drivers >/dev/null)
+$(shell mkdir -p $(DEPDIR)/hooks >/dev/null)
 
 # e.g., 4.9.3 -> 40903
 CXXVERSION := $(shell $(CXX) -dumpversion | sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g' -e 's/^[0-9]\{3,4\}$$/&00/')
-
 CXXVERSION_4_8_OR_HIGHER := $(shell expr $(CXXVERSION) \>= 40800)
-ifeq "$(CXXVERSION_4_8_OR_HIGHER)" "0"
-  $(error g++ 4.8 or higher is required)
+
+IS_GCC := $(shell expr "$(shell $(CXX) --version)" : "g++")
```

Comment:
Minor nit: perhaps we can use something like `CXXCOMPILER := $(word 1, $(shell $(CXX) --version))`, so that the value can be either "g++" or "clang"? The name `IS_GCC` implies it is boolean, i.e., 0 or 1, while it's not. Also, while it's very unlikely, in a future version `clang++ --version` may print out "clan**g++**" somewhere in the message 😛